xarray/tests/test_datatree.py

```diff
@@ -1793,6 +1834,42 @@ def test_arithmetic_inherited_coords(self):
         expected["/foo/bar"].data = np.array([8, 10, 12])
         assert_identical(actual, expected)
 
+    def test_binary_op_commutativity_with_dataset(self):
+        # regression test for #9365
+
+        ds1 = xr.Dataset({"a": [5], "b": [3]})
+        ds2 = xr.Dataset({"x": [0.1, 0.2], "y": [10, 20]})
+        dt = DataTree.from_dict(
+            {
+                "/": ds1,
+                "/subnode": ds2,
+            }
+        )
+
+        other_ds = xr.Dataset({"z": ("z", [0.1, 0.2])})
+
+        expected = DataTree.from_dict(
+            {
+                "/": ds1 * other_ds,
+                "/subnode": ds2 * other_ds,
+            }
+        )
+
+        result = other_ds * dt
+        assert_equal(result, expected)
+
+    def test_inplace_binary_op(self):
+        ds1 = xr.Dataset({"a": [5], "b": [3]})
+        ds2 = xr.Dataset({"x": [0.1, 0.2], "y": [10, 20]})
+        dt = DataTree.from_dict({"/": ds1, "/subnode": ds2})
+
+        expected = DataTree.from_dict({"/": ds1 + 1, "/subnode": ds2 + 1})
+
+        dt += 1
+        assert_equal(dt, expected)
+
+    # TODO test single-node datatree doesn't broadcast
```

Comment:
Will also just do this in a follow-up as the implementation is a change to `map_over_subtree`, so really orthogonal to this PR.

Reply:
Actually turns out there was nothing to change here - it already behaves the way we want it to. I added a test in https://github.com/pydata/xarray/pull/9619/commits/28148019b07e4471cea0862adf31ab8d46c7a902, and this answers the comment https://github.com/pydata/xarray/issues/9365#issuecomment-2291622577.